src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -1023,33 +1023,36 @@ namespace AirLoopHVACDOAS {
         int constexpr winterDesignDayTypeIndex(10);
 
         for (size_t i = 1; i <= state.dataWeatherManager->DesDayInput.size(); i++) {
+            auto &desDayInput = state.dataWeatherManager->DesDayInput(i);
             // Summer design day
-            if (state.dataWeatherManager->DesDayInput(i).DayType == summerDesignDayTypeIndex && state.dataAirLoopHVACDOAS->SummerDesignDayFlag) {
-                this->SizingCoolOATemp = state.dataWeatherManager->DesDayInput(i).MaxDryBulb;
-                if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::WetBulb) { // wet bulb temperature
-                    this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdbTwbPb(
-                        state, this->SizingCoolOATemp, state.dataWeatherManager->DesDayInput(i).HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                } else if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::DewPoint) { // dewpoint
-                    this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdpPb(
-                        state, state.dataWeatherManager->DesDayInput(i).HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                } else if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::HumRatio) {
-                    this->SizingCoolOAHumRat = state.dataWeatherManager->DesDayInput(i).HumIndValue;
-                } // else { // What about other cases?
-                state.dataAirLoopHVACDOAS->SummerDesignDayFlag = false;
```

Comment:
This flag here, state.dataAirLoopHVACDOAS->SummerDesignDayFlag, and same for winter design day, found first design day object data and then disregarded the rest (because this global flag is included in the IF conditional test). If there were any other DOAS systems the design conditions could not be found/initialized.